pkg/kubefedctl/join_test.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubefedctl
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/http/httputil"
+	"net/url"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	utiltesting "k8s.io/client-go/util/testing"
+
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/util"
+
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+)
+
+var (
+	joiningClusterName = "joiningcluster"
+	hostClusterName    = "hostnamecluster"
+)
+
+var _ = Describe("Kubefedctl", func() {
+	BeforeEach(func() {
```

Comment:
Same question as above? Please remove the empty functions if not of intentional use.

Reply:
I kept them to have them as a skeleton, they didn't bother in my opinion. I'll remove them.